dev/generate_bom.py

```diff
@@ -83,6 +85,8 @@ def init_argument_parser(cls, parser):
                         help="Docker registry to push the container images to.")
     parser.add_argument('--changelog_output', default='',
                         help="Output file to write the changelog to.")
+    parser.add_argument('--nightly', default=False, action='store_true',
```

Comment:
I was getting at "nightly" is an external policy, not part of this tool. Either the caller would alias the specific bom as a "nightly" or it would ask the tool to as you've done here. But again, "nightly" seems very one-off. This should be a means to ask for an alias. The default is None but the caller can pass in "nightly" -- or anything else.

I prefer having the tooling such that the caller can do this after the fact. That way you can have something more useful like "stable" (passed automated tests but not yet approved for release, or even release candidates for new major releases that might involve a communal approval process) which cannot be determine at the time this is called. Heck, you could even rollback a symbolic alias to point to an older version if you discover a problem after you set it.

It could be that this tool does the aliasing, but should be after the fact. You ask the tool to alias a specific version as another name. The reason for that is so the knowledge is encapsulated in this one tool. The control is left external as to when, why and what to alias as. Not a narrow special case hard wired into the tool. The special case is more efficient and convenient, but not by much. We're adding this switch to save one call from one client as opposed to support a mechanism that can be used for a more general range of use cases covering this special case as well as other more interesting and useful ones.